### DISCUSSION TITLE
Not able to provide custom error message for MUI Autocomplete control

### DISCUSSION THREAD

**Discussion by neumadhu** [colinhacks/zod#3297] [Q&A] [answered]
I am using MUI Autocomplete + React hook form + Zod for the validation. When I tried to apply validation to the MUI control I am getting "Expected object, received null" message when I use the validation rule like  fieldName: z.object({}),

![image](https://github.com/colinhacks/zod/assets/75621605/93ed7d56-df9c-433b-b57a-258aa9c6eeb0)

But when I use fieldName: z.string(), it is throwing below error ("Expected string, received object")even though I select the item.
![image](https://github.com/colinhacks/zod/assets/75621605/027eb0b8-6a98-46bc-b0f4-1ce567ae2fe6)

Please let me know How can I resolve it.

**Comment by JacobWeisenburger** [comment, marked as answer]
Please send a full reproducible code example.
